frontend/ui/elements/screen_android.lua

```diff
@@ -0,0 +1,61 @@
+local isAndroid, android = pcall(require, "android")
+local Device = require("device")
+local Geom = require("ui/geometry")
+local logger = require("logger")
+local Input = Device.input
+local Screen = Device.screen
+
+if not isAndroid then return end
+
+-- TODO: refactor this code and call it from android/device.lua, to restore last saved state
+-- at launch instead of starting always in fullscreen.
+
+local ScreenHelper = {}
+
+local function flipFullscreen()
+    -- toggle android status bar visibility
+    local is_fullscreen = android.isFullscreen()
+    android.setFullscreen(not is_fullscreen)
+    logger.dbg(string.format("requesting fullscreen: %s", not is_fullscreen))
+
+    local screen_width = android.getScreenWidth()
+    local screen_height = android.getScreenHeight()
+    local status_bar_height = android.getStatusBarHeight()
+    local new_height = screen_height - status_bar_height
+
+    if not is_fullscreen and Screen.viewport then
+        status_bar_height = 0
+        -- reset touchTranslate to normal
+        Input:registerEventAdjustHook(
+            Input.adjustTouchTranslate,
+            {x = 0 + Screen.viewport.x, y = 0 + Screen.viewport.y})
+    end
+
+    local viewport = Geom:new{x=0, y=status_bar_height, w=screen_width, h=new_height}
+    logger.info(string.format("Switching viewport to new geometry [x=%d,y=%d,w=%d,h=%d]",
+        0, status_bar_height, screen_width, new_height))
+
+    Screen:setViewport(viewport)
+
+    if is_fullscreen and Screen.viewport then
+        Input:registerEventAdjustHook(
+            Input.adjustTouchTranslate,
+            {x = 0 - Screen.viewport.x, y = 0 - Screen.viewport.y})
+    end
+end
+
+local function flipWakelock()
+    is_wakelock = not G_reader_settings:isTrue("disable_android_wakelock")
```

Comment:
What's the intended scope for this variable? Since it looks to be used only here, it should be:

```suggestion
    local is_wakelock = not G_reader_settings:isTrue("disable_android_wakelock")
```